x/account/types/genesis.go

```diff
@@ -23,5 +25,32 @@ func (gs GenesisState) Validate() error {
 
 	// this line is used by starport scaffolding # genesis/types/validate
 
+	// Check for duplicated index in coordinatorByAddress
+	coordinatorByAddressIndexMap := make(map[string]uint64)
+	for _, elem := range gs.CoordinatorByAddressList {
+		index := string(CoordinatorByAddressKey(elem.Address))
+		if _, ok := coordinatorByAddressIndexMap[index]; ok {
+			return fmt.Errorf("duplicated index for coordinatorByAddress")
+		}
+		coordinatorByAddressIndexMap[index] = elem.CoordinatorId
+	}
+
```

Comment:
Can we create `genesis_test.go` to test it correctly fails if there is inconsistencies with `Coordinator` and `CoordinatorByAddress`?

We don't generate one with the command since duplication check is a bit trivial but uit may be good for the future to always create tests when we have some more advance logic with genesis

Reply:
I created one for the `launch` module if you want some inspiration: https://github.com/tendermint/spn/blob/feat/init-genesis-account/x/launch/types/genesis_test.go